lib/middleware.js

```diff
@@ -1,10 +1,61 @@
 'use strict';
 
+require('dotenv').load();
+const jwt = require('jwt-simple');
+
 module.exports = function(req, res, cb) {
-  /* To be replaced with token validation and/or authentication */
-  const validated = Object.assign(req, {
-    user: 'test-user',
-    isAdmin: false
-  });
-  return cb(null, validated);
+  const token = getToken(req);
+  let payload;
+  if (token) {
+    payload = getPayload(token);
+  } else {
+    payload = null;
+  }
+
+  // QUESTION: should we check if the user if belongs to the list
+  // of authorized users here and add like an `.isTrustedUser` property or
+  // so to the request?
+  if (payload) {
+    req.user = payload;
+    if (process.env.trustedUsers) {
+      const trustedUsers = JSON.parse(process.env.trustedUsers);
+      if (trustedUsers.indexOf(payload.id) !== -1) {
+        req.isTrustedUser = true;
+      } else {
+        req.isTrustedUser = false;
+      }
```

Comment:
Using ternary operators may make this line too long, but you could also write this as:

```js
req.isTrustedUser = trustedUsers.indexOf(payload.id) !== -1 ? true : false;
```

Reply:
aha, but this could be further shortened to `req.isTrustedUser = trustedUsers.indexOf(payload.id) !== -1` :-)